### ISSUE TITLE
prefix.cc acting up on IRIs

### ISSUE THREAD

**Issue by VladimirAlexiev** [cygri/prefix.cc#17]
I'm trying to add a prefix 
          bgcat http://bg.dbpedia.org/resource/Категория:

prefix.cc says "URI must start with http:// or https://, end in / or : or #."
But it does (ends in ":")!
Maybe it's disagreeing with the Cyrilllic characters?

Richard: I did a few tests and you're right, it's the non-URI characters.

**Comment by cygri** [comment]
A bit of progress towards fixing this.

- I've tightened up the regex for namespace URIs, and made it more readable (see [source](https://github.com/cygri/prefix.cc/blob/f105b7e64ca0c93e2358056f860b195f993669b0/lib/namespaces.class.php#l135)).
- I've added a [little test script](https://github.com/cygri/prefix.cc/blob/master/tests/tests.php) that can be run on the command line.

There's an [ABNF for IRIs in RFC 3987](https://tools.ietf.org/html/rfc3987#section-2.2). The problem is that the `ucschar` seen in the `iunreserved` production is not reflected in the `$subdomain` and `$path` parts of the prefix.cc regex. So what needs to be done:

- Modify the regex to allow the full `ucschar` range, assuming UTF-8 encoding. PHP's `preg_match` [has a `u` modifier](http://php.net/manual/en/reference.pcre.pattern.modifiers.php) that should be helpful here.
- Conveniently, the database column for storing namespace URIs already is UTF-8 capable, so nothing to do there.
- Check that the `prefix.cc/foaf.go` style and reverse lookups work correctly; some URI-to-IRI or IRI-to-URI conversion may be necessary otherwise.
- Perhaps make a pass over source code and output strings to make sure we say “IRI” where we should.
